lyse/__main__.py

```diff
@@ -2328,19 +509,15 @@ def delete_items(self, confirm):
 
 
 if __name__ == "__main__":
-    logger = setup_logging('lyse')
-    labscript_utils.excepthook.set_logger(logger)
-    logger.info('\n\n===============starting===============\n')
+
+    splash.update_text('starting GUI')
     qapplication = QtWidgets.QApplication.instance()
     if qapplication is None:
         qapplication = QtWidgets.QApplication(sys.argv)
     qapplication.setAttribute(QtCore.Qt.AA_DontShowIconsInMenus, False)
-    app = Lyse()
 
-    # Start the web server:
-    splash.update_text('starting analysis server')
-    server = WebServer(app.port)
-    splash.update_text('done')
```

Comment:
The relocation of the `WebServer` into the `Lyse` class, and instantiating it before the GUI is set up, means that any messages received immediately on start could crash lyse when the `WebServer.handler` method tries to access something about the UI that isn't instantiated yet.

The previous implementation worked such that messages could be received, and placed in appropriate event queues before the Qt loop was even started.

I suspect, given the auto-retry behaviour of BLACS, that there would be scenarios where the crash will occur (it's a bit of a race condition though so may not be easily replicable)